### PR TITLE
Add PowerShell formatter linter

### PR DESCRIPTION
**Pull request by bdovaz**
@nvuillam I put it in draft because I have the doubt of how to create a test that fails if this command never fails... I pass it an input that is not even a valid powershell code and it works, that is, it does nothing and returns the same input as output....

https://learn.microsoft.com/en-us/powershell/module/psscriptanalyzer/invoke-formatter?view=ps-modules

Formatter code that does not handle any errors:

https://github.com/PowerShell/PSScriptAnalyzer/blob/1.21.0/Engine/Formatter.cs

Unlike ScriptAnalyzer.cs which does handle errors (used by original powershell linter):

https://github.com/PowerShell/PSScriptAnalyzer/blob/1.21.0/Engine/ScriptAnalyzer.cs

### CODE REVIEW

**Comment by nvuillam** [comment]
Don't forget to build with `bash build.sh`, it will generate the test class :)

**Comment by bdovaz** [comment]
@nvuillam done.

Can you read and answer me to the first message of this PR related to the tests?

I see that there are special cases where you exclude this type of test (linter_failure):

https://github.com/oxsecurity/megalinter/blob/v6.16.0/megalinter/tests/test_megalinter/helpers/utilstest.py#L214

If we don't find a solution I don't know if this would be the solution.

**Comment by nvuillam** [comment]
@bdovaz do powershell formatter have a --check mode, to just check if the file is already formatted or not ?

**Comment by bdovaz** [comment]
@nvuillam as you can see, no:

https://learn.microsoft.com/en-us/powershell/module/psscriptanalyzer/invoke-formatter?view=ps-modules

**Comment by nvuillam** [comment]
@bdovaz it seems that Microsoft does not like to build their apps like all others ^^

In that case you can add an empty file named `no_test_failure` (without extension) in the test folder

File: megalinter/descriptors/powershell.megalinter-descriptor.yml [hunk 1]
```diff
@@ -53,3 +53,25 @@ linters:
       vscode:
         - name: VSCode PowerShell extension
           url: https://marketplace.visualstudio.com/items?itemName=ms-vscode.PowerShell
+  # Internal powershell linter
+  - class: PowershellLinter
+    linter_name: powershell_formatter
+    linter_url: https://github.com/PowerShell/PSScriptAnalyzer
```
Comment (nvuillam) [inline megalinter/descriptors/powershell.megalinter-descriptor.yml:59]: @bdovaz plz add `    is_formatter: true` here :)

**Review by nvuillam — CHANGES_REQUESTED** [review]
plz check comment :)

File: megalinter/descriptors/powershell.megalinter-descriptor.yml (hunk 1 above)
Comment (bdovaz) [reply megalinter/descriptors/powershell.megalinter-descriptor.yml:60]: Done

**Comment by nvuillam** [comment]
```shell
❌ Linted [REPOSITORY] files with [git_diff]: Found 1 error(s) - (0.03s)
- Using [git_diff v2.36.3] https://megalinter.io/2644adfa201ad6bd9bec5f21e8040cd699416064/descriptors/repository_git_diff
- MegaLinter key: [REPOSITORY_GIT_DIFF]
- Rules config: identified by [git_diff]
--Error detail:
.automation/test/powershell_formatter/good/powershell_good_1.ps1:2: new blank line at EOF.
.automation/test/powershell_formatter/good/powershell_good_1.psd1:4: new blank line at EOF.
.automation/test/powershell_formatter/good/powershell_good_1.psm1:2: new blank line at EOF.
```

That's really strange, I never saw this one :D 
Maybe some error while mananing conflicts ?

**Comment by bdovaz** [comment]
> ```shell
> ❌ Linted [REPOSITORY] files with [git_diff]: Found 1 error(s) - (0.03s)
> - Using [git_diff v2.36.3] https://megalinter.io/2644adfa201ad6bd9bec5f21e8040cd699416064/descriptors/repository_git_diff
> - MegaLinter key: [REPOSITORY_GIT_DIFF]
> - Rules config: identified by [git_diff]
> --Error detail:
> .automation/test/powershell_formatter/good/powershell_good_1.ps1:2: new blank line at EOF.
> .automation/test/powershell_formatter/good/powershell_good_1.psd1:4: new blank line at EOF.
> .automation/test/powershell_formatter/good/powershell_good_1.psm1:2: new blank line at EOF.
> ```
> 
> That's really strange, I never saw this one :D Maybe some error while mananing conflicts ?

Well I don't know what to do to fix this because if you look at my PR I simply copied the 3 files from the powershell folder as is.

**Comment by nvuillam** [comment]
@bdovaz what if you add a blank line in the files so they errors can disappear ?

**Comment by bdovaz** [comment]
@nvuillam I just did it but that is the solution? I just don't understand the error either, what it refers to and how to fix it.

Let's wait for the result of the build.

**Comment by nvuillam** [comment]
That's really really strange... as the rest seems legit,  I force the PR, we'll see if it solves the issue ^^

**Comment by echoix** [comment]
This reminds me of when git changes the end of lines when checking-out, and changing them back when committing. It is normally controlled by a .gitattributes file (like https://github.com/oxsecurity/megalinter/blob/main/.gitattributes ). Are you working on Windows?

**Comment by bdovaz** [comment]
> This reminds me of when git changes the end of lines when checking-out, and changing them back when committing. It is normally controlled by a .gitattributes file (like https://github.com/oxsecurity/megalinter/blob/main/.gitattributes ). Are you working on Windows?

@nvuillam @echoix yes, I'm on Windows.

**Comment by nvuillam** [comment]
I'm on windows too but never had the issue ^^

If it does not happen on this PR https://github.com/oxsecurity/megalinter/pull/2189 , let's consider the problem solved ^^

**Comment by nvuillam** [comment]
I learned something today :)

https://stackoverflow.com/questions/27059239/git-new-blank-line-at-eof

**Comment by echoix** [comment]
> I'm on windows too but never had the issue ^^

Some of the .gitattributes won't get applied to an already cloned repo without any actions, so it doesn't exclude the problem. 

https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings

https://git-scm.com/docs/gitattributes

**Comment by nvuillam** [comment]
Finally solved... i just remove the line at the end of the files >_<

**Pull request merged**